govcd/metadata.go

```diff
@@ -12,6 +12,18 @@ import (
 	"github.com/vmware/go-vcloud-director/v2/types/v56"
 )
 
+// MetadataCompatible allows consumers of this library to consider all structs that implement
+// this interface to be the same type
+type MetadataCompatible interface {
```

Comment:
Interesting read! Then probably the best option here is to declare a private interface in `metadata_test` and move this one to https://github.com/vmware/terraform-provider-vcd/pull/853

Reply:
About the capitalization, in this case `Vdc` is used as it's the type name (that way the IDE can show its contents in contextual menu and link to it). Not sure if you were referring to that one?

Reply:
> About the capitalization, in this case Vdc is used as it's the type name (that way the IDE can show its contents in contextual menu and link to it). Not sure if you were referring to that one?

I don't think so. Main place to use VDC capitalization is in user-facing messages. Actually, @Didainius wrote that as well:

> A few notes on capitalization in comments 

:)

Reply:
I can change it, but we'd lose this feature in the IDE (not sure about others):
![Screenshot 2022-05-25 at 15 49 44](https://user-images.githubusercontent.com/3122132/170278348-33f45593-e181-4e4c-b241-1df9cf582be4.png)

Here's the result fully with caps:
![Screenshot 2022-05-25 at 15 49 52](https://user-images.githubusercontent.com/3122132/170278399-144c6985-371f-4c04-8dd1-008789ce0306.png)

Although is also true that the type is already in the receiver object `func (vdc *Vdc)...` so that would be not a big loss neither.

I'll change them if you agree with that.

Reply:
Done, `Vdc` is now `VDC` in most of the comments.
I left hose in lowercase:
```
// Deprecated: use Vdc.AddMetadataEntryAsync.
```
Because in here it's indeed referring to the type `Vdc`.

Reply:
Could not find lower case `org` in comments, though.